### PR TITLE
Add Windows OS Product Type to User Agent string

### PR DESCRIPTION
**Pull request by jadeiceman**
Add Windows OS Product Type to User Agent string

## Checklist
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [X] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [X] This pull-request is submitted against the `master` branch.

## Description of the changes
- Add Windows OS product type to user agent string to differentiate/identify Windows SKUs in telemetry
- Update existing user agent test to include product type

## Reference/Link to the issue solved with this PR (if any)
Addresses internal task/request for Windows OS SKU information to be included in user agent

### CODE REVIEW

File: iothub/device/src/NativeMethods.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) &&
+                    GetProductInfo(Environment.OSVersion.Version.Major, Environment.OSVersion.Version.Minor, 0, 0, out int productType))
+                {
+                    return productType;
+                }
+            }
+            catch (DllNotFoundException)
+            {
+                // Catch any DLL not found exceptions
```
Comment (CIPop) [inline iothub/device/src/NativeMethods.cs]: Please add Debug.Assert and error logging when this exception is thrown.

**Comment by CIPop** [comment]
Scheduled https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_build/results?buildId=4052&view=logs

File: iothub/device/src/NativeMethods.cs (hunk 1 above)
Comment (jadeiceman) [reply iothub/device/src/NativeMethods.cs]: Sure, added.  Let me know if it looks right.  I just mimic'ed from what I could find in the rest of the code.

**Comment by jadeiceman** [comment]
@CIPop I've fixed the build breaks, could you run the CI build again?  Thanks!

**Comment by CIPop** [comment]
@leantk PTAL /AzurePipelines command above. (I saw it worked for you.)

**Comment by CIPop** [comment]
https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_build/results?buildId=4640&view=logs

**Pull request merged**